src/core/meshcop/dataset_manager.cpp

```diff
@@ -330,14 +330,20 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
     Tlv::Type type;
     bool isUpdateFromCommissioner = false;
     bool isUpdateAffectConnectivity = false;
+    bool isUpdateIdenticalValue = true;
```

Comment:
Maybe combing two bool variables is a little hard to achieve. We need to iterate the MGMG_ACTIVE_SET request message first to find if there is any tlv which affects connectivity (if so, `isUpdateAffectConnectivity` flag is true). Then after the iteration finishes, check if any value is different from its own (if so, `isUpdateIdenticalValue` is false). If we combine the two bool variables, for example, we take advantage of this combined variable to iterate the request message, there will be one case happened, is that when we do not iterate to some tlv (the tlv value is not read from message), but in the same loop we need to check if its value is different current one, at this moment the check will be true since we do not get the specific tlv value before, `isUpdateIdenticalValue` will be false. This will cause an incorrect result.